maestro/demos/CONTRIBUTING.md

```diff
@@ -0,0 +1,185 @@
+# Contributing to Bee Hive Demos
+
+Thank you for your interest in contributing to **Bee Hive**! This document outlines the structure and workflow for setting up and testing demos using **Maestro**.
+
+---
+
+## Project Structure
+
+```markdown
+/maestro
+│── /src                    # Source code for core functionalities
+│   │── agent.py            # Base class for defining agents
+│   │── bee_agent.py        # Implementation of BeeAgent
+│   │── create_agents.py    # Creates agents from YAML definitions
+│   │── run_workflow.py     # Executes workflows sequentially
+│   │── step.py             # Step execution logic for workflows
+│   │── /schemas            # JSON schema definitions for agents and workflows
+│       │── agent_schema.json
+│       │── workflow_schema.json
+│── /demos                  # Demo-specific workflows and agents
+│   │── /workflows          
+│       │── agents.yaml     # YAML file defining the agents used in the demo
+│       │── workflow.yaml         
+│── .env                    # Environment variables (use `example.env` as reference)
+│── README.md               # Project documentation
+│── CONTRIBUTING.md         # Contribution guidelines
+```
+
+---
+
+## Setting Up Your Environment
+
+### 1. Clone the Repository
+
+```bash
+git clone https://github.com/i-am-bee/bee-hive.git
+cd bee-hive
+```
+
+### 2. Set Up the Virtual Environment
+
+```bash
+poetry env use python3
+poetry install
+poetry shell
+```
+
+### 3. Configure Environment Variables
+
+- Copy `example.env` and rename it to `.env`:
+
+  ```bash
+  cp example.env .env
+  ```
+
+---
+
+## Validating and Running Demos with Maestro
+
+### Validate Workflow and Agent YAML Files
+
+Note: All these commands assume you are running from maestro top-level directory.
+
+Ensure that your YAML definitions adhere to the correct schema before running them.
+
+```bash
+maestro validate ./schemas/agent_schema.json ./src/agents/meta_agent/agents.yaml
+```
+
+### Create Agents
+
+To create agents (with the ability to manually add tools):
+
+```bash
+maestro create ./src/agents/meta_agent/agents.yaml
+```
+
+### Run the Workflow
+
+If you have already created the agents and enabled the tools:
+
+```bash
+maestro run None ./src/agents/meta_agent/workflow.yaml
+```
+
+OR run the workflow directly:
```

Comment:
Add something to the effect that running workflow with agents file will create agent if they are not already